### PR TITLE
feat: implement prediction pynative model recipe

### PR DESCRIPTION
**Pull request by shekhar-rudder**
## Description of the change

This PR implements the prediction pynative model
- Extracted out the common predict code so that it can be used by both python and pynative models
- Added 2 proxy functions in pyNativeWht
- Created common util code for getting path of model_configs.yaml file

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/rudderstack_predictions/ml_core/preprocess_and_predict.py [hunk 1]
```diff
@@ -59,7 +59,6 @@ def preprocess_and_predict(
     categorical_columns = results["column_names"]["feature_table_column_types"][
         "categorical"
     ]
-    arraytype_columns = results["column_names"]["input_column_types"]["arraytype"]
```
Comment (shekhar-rudder) [inline src/predictions/rudderstack_predictions/ml_core/preprocess_and_predict.py:62]: unused variable

File: scripts/simulate_py_native.py [hunk 2]
```diff
@@ -25,24 +28,20 @@ def de_ref(self, model_ref):
...
+        "ml_config": {
+            "data": {"label_column": "is_churned_7_days", "prediction_horizon_days": 7}
+        },
     }
+    homedir = os.path.expanduser("~")
+    with open(os.path.join(homedir, ".pb/siteconfig.yaml"), "r") as f:
+        creds = yaml.safe_load(f)["connections"]["test"]["outputs"]["redshift"]
     input_model_refs = build_spec.get("inputs", [])
-    output_filename = os.path.join(os.getcwd(), "training_file")
+    output_filename = "/Users/shashankshekhar/Documents/RudderStack/rudderstack-profiles-classifier/samples/py_native/output/redshift/seq_no/1093/Material_training_model_9b28e42b_training_file"
```
Comment (dpatchigolla) [inline scripts/simulate_py_native.py:39]: Can you make this path more generic? 
And can this file run in others' machines, or does it need any local files?

**Review by dpatchigolla — APPROVED** [review]
minor clarifications

File: predict.py [hunk 3]
```diff
@@ -1,34 +1,12 @@
...
-from src.predictions.rudderstack_predictions.trainers.MLTrainer import (
-    ClassificationTrainer,
-    RegressionTrainer,
+sys.path.append(
+    os.path.abspath(os.path.join(os.path.dirname(__file__), "src", "predictions"))
 )
 
-warnings.filterwarnings("ignore", category=NumbaDeprecationWarning)
-warnings.simplefilter("ignore", category=NumbaPendingDeprecationWarning)
+from rudderstack_predictions.predict import _predict
```
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply predict.py:9]: Added a comment explaining the choice

File: scripts/simulate_py_native.py (hunk 2 above)
Comment (shekhar-rudder) [reply scripts/simulate_py_native.py]: Reverted this change. Should work for all the machines

File: src/predictions/rudderstack_predictions/py_native/training.py [hunk 4]
```diff
@@ -112,3 +119,8 @@ def execute(self, this: WhtMaterial):
             PyNativeWHT(this),
             constants.ML_CORE_PYNATIVE_PATH,
         )
+        # pb expects every model to create a material. The output material of training model is a file which is currently not supported
```
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/rudderstack_predictions/py_native/training.py:122]: Created a [ticket](https://linear.app/rudderstack/issue/PRML-469/use-training-material-table-instead-of-training-metrics-v4-to-write). Will pick it up next

File: predict.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply predict.py]: Changed to `src.predictions....`

**Pull request merged**